cmd/flat-network-controller/app/controller.go

```diff
@@ -150,15 +159,21 @@ func (c *Controller) process(item interface{}) {
 }
 
 // manageFlatNetwork is the core logic to manage a flat typed network object
-func manageFlatNetwork(net *v1.Network, netClient arktos.Interface, svcClient kubernetes.Interface) error {
+func manageFlatNetwork(net *v1.Network, netClient arktos.Interface, svcClient kubernetes.Interface, toDeployDNS bool, domainName string) error {
 	if len(net.Status.DNSServiceIP) != 0 {
 		klog.V(5).Infof("network %s/%s/%s already have DNS service IP %s; skipped", net.Tenant, net.Namespace, net.Name, net.Status.DNSServiceIP)
 		return nil
 	}
 
 	svc, err := createOrGetDNSService(net, svcClient)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get or create per-network DNS service: %v", err)
```

Comment:
It may be helpful to log the tenant name in the error message.

Reply:
it makes sense. will update as suggested.